### PR TITLE
Async build typos and miscellaneous fixes

### PR DESCRIPTION
**Pull request by lealem47**
# Description
Fix for async build with enable all and the following warnings:

```
wolfcrypt/test/test.c:667:22: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Werror,-Wint-conversion]
        errno_string = strerror_r(WC_TEST_RET_DEC_I(es),
```

```
wolfcrypt/src/siphash.c:638:9: error: string literal of length 6770 exceeds maximum length 4095 that ISO C99 compilers are required to support [-Werror,-Woverlength-strings]
        "ldp    x12, x13, [%[key]]\n\t"
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Fixes zd#

# Testing

Make check and multi test

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Please check:

```
--enable-aesgcm=table --enable-all --enable-intelasm --enable-sp-math-all 
```

and 

```
--enable-valgrind
```

Though the --enable-valgrind may be a false positive

**Review by dgarske — APPROVED** [review]

**Comment by lealem47** [comment]
> Please check:
> 
> ```
> --enable-aesgcm=table --enable-all --enable-intelasm --enable-sp-math-all 
> ```
> 
> and
> 
> ```
> --enable-valgrind
> ```
> 
> Though the --enable-valgrind may be a false positive

Both are clean locally and jenkins tests pass on rerun (PRB-generic-config-parser build #471)

**Review by JacobBarthelmeh — APPROVED** [review]
Thanks for checking the build configurations

**Review by douzzer — APPROVED** [review]
LGTM.  tests good locally on `wolfssl-multi-test.sh ... super-quick-check`.

File: wolfcrypt/src/siphash.c [hunk 1]
```diff
@@ -744,6 +744,12 @@ int wc_SipHash(const unsigned char* key, const unsigned char* in, word32 inSz,
         "cmp    %w[outSz], #8\n\t"
         "b.eq   L_siphash_8_end\n\t"
 
+    : [in] "+r" (in), [inSz] "+r" (inSz)
```
Comment (dgarske) [inline wolfcrypt/src/siphash.c:747]: Sean, this broke the ARMv8 with `--enable-debug`. Can you review this section for @lealem47 ?

Comment (SparkiDev) [reply wolfcrypt/src/siphash.c:747]: Looking into it.
I will put up.a PR that doesn't break --enable-debug on M1.

Comment (SparkiDev) [reply wolfcrypt/src/siphash.c:747]: Fix: #6246

**Pull request merged**